pyproject.toml

```diff
@@ -62,5 +58,5 @@ ignore_names = ["pytest_*"]
 paths = ["pytest_socket.py"]
 
 [build-system]
-requires = ["poetry-core>=2.0"]
-build-backend = "poetry.core.masonry.api"
+requires = ["hatchling"]
+build-backend = "hatchling.build"
```

Comment:
There's a newer `uv_build` backend - more context here: https://docs.astral.sh/uv/concepts/build-backend/

Let's try to use the new `uv_build` backend if possible, and only revert back to hatchling if we cannot produce distribution archives the same as poetry does currently.

Reply:
That's good news - let's simplify our lives by updating the project structure to be a src/based layout and thereby allow usage of `uv_build` is we can still do a custom include.

Reply:
The `common` test paths should be explicitly relative paths, so that installing `https://pypi.org/project/common/` doesn't override the import paths, as well as confirm to isort rules correctly.

There's one pre=commit tool for .editorconfig that didn't run, and there's a failure that needs to be fixed.